api/types.go

```diff
@@ -31,17 +31,23 @@ func (e StatusError) Error() string {
 }
 
 type GenerateRequest struct {
-	Model    string `json:"model"`
-	Prompt   string `json:"prompt"`
-	System   string `json:"system"`
-	Template string `json:"template"`
-	Context  []int  `json:"context,omitempty"`
-	Stream   *bool  `json:"stream,omitempty"`
-	Raw      bool   `json:"raw,omitempty"`
+	Model    string    `json:"model"`
+	Prompt   string    `json:"prompt"`
+	System   string    `json:"system"`
+	Template string    `json:"template"`
+	Context  []int     `json:"context,omitempty"`  // DEPRECATED: context is deprecated, use messages instead
+	Messages []Message `json:"messages,omitempty"` // messages sent in the conversation so far
+	Stream   *bool     `json:"stream,omitempty"`
+	Raw      bool      `json:"raw,omitempty"`
 
 	Options map[string]interface{} `json:"options"`
 }
 
+type Message struct {
+	Prompt   string `json:"prompt"`
+	Response string `json:"response"`
+}
```

Comment:
```suggestion
type Message struct {
	Role   string `json:"role"`
	Content string `json:"content"`
}
```

Here was a (very quick and dirty) example of how to template out a series of messages using the existing Ollama templates:https://github.com/jmorganca/ollama/commit/949fc4eafa0cabacb2bf643d1c90a51a45ccc4a4#diff-782c2737eecfa83b7cb46a77c8bdaf40023e7067baccd4f806ac5517b4563131R57

Reply:
I prefer 

```
type Message struct {
	Prompt   string `json:"prompt"`
	Response string `json:"response"`
}
```

But it doesn't really matter, each format can be mapped to the other format interchangeably.